### ISSUE TITLE
Error: Algolia failed: ID is not defined

### ISSUE THREAD

**Issue by Edmund1645** [u12206050/gridsome-plugin-algolia#6]
Hello, thanks a lot for this plugin, I recently just added a `modified` field in my frontmatter so i could use partial updates in the plugin, but when I try to build, I get this error: 
```bash
Algolia collection #0: Executing query
Algolia collection #0: items in collection 5
Algolia collection #0: starting Partial updates
Algolia collection #0: found 6 existing items
Error: Algolia failed: ID is not defined at afterBuild (/home/edmund/projects/projects/my own/NinjaBlog/node_modules/gridsome-plugin-algolia/index.js:184:13)
```

This is how my config looks: 
```js
const collections = [
  {
    query: `{
      allPost {
        edges {
          node {
            id
            title
            path
            modified
          }
        }
      }
    }`,
    transformer: ({ data }) => data.allPost.edges.map(({ node }) => node),
    indexName: process.env.ALGOLIA_INDEX_NAME, // Algolia index name
    itemFormatter: item => {
      return {
        objectID: item.id,
        title: item.title,
        slug: item.path,
        modified: String(item.modified)
      };
    }, // optional
    matchFields: ['modified'] // Array<String> required with PartialUpdates
  }
];



module.exports = {

  plugins: [
 {
      use: `gridsome-plugin-algolia`,
      options: {
        appId: process.env.ALGOLIA_APP_ID,
        apiKey: process.env.ALGOLIA_ADMIN_KEY,
        collections,
        chunkSize: 10000, // default: 1000
        enablePartialUpdates: true // default: false
      }
]
  }

```
I have my `.env` file in place and all keys provided there. 

I'm currently using the version `2.1.2` of the plugin.

**Comment by u12206050** [comment]
Try putting a console.log within the itemFormatter to make sure that item
is valid.

**Comment by Edmund1645** [comment]
This is the data: 
```js
[Object: null prototype] {
  id: '27d77fce22e5ce3dbd3915ebeee80f5b',
  title:
   'How to destructure arrays and objects in JavaScript - part 2',
  path:
   '/how-to-destructure-arrays-and-objects-in-java-script-part-2/',
  modified: '2019-10-19T00:00:00.000Z' }
[Object: null prototype] {
  id: 'b7bc3f13e905da9e93de90d0e5c439cc',
  title:
   'How to destructure arrays and objects in JavaScript - part 1',
  path:
   '/how-to-destructure-arrays-and-objects-in-java-script-part-1/',
  modified: '2019-10-15T00:00:00.000Z' }
[Object: null prototype] {
  id: '6a4fa5b0d147669f0b4370e656365a7a',
  title: 'Advanced styling with CSS attribute selectors',
  path: '/advanced-styling-with-css-attribute-selectors/',
  modified: '2019-10-11T00:00:00.000Z' }
[Object: null prototype] {
  id: 'a616fca2ef90b306dda49ff1281d691a',
  title: 'How to use arrow functions in JavaScript.',
  path: '/how-to-use-arrow-functions-in-java-script/',
  modified: '2019-09-09T00:00:00.000Z' }
[Object: null prototype] {
  id: 'a19a89aa18a7d14cbc24e8e51f658159',
  title: 'Five tips to master a programming language.',
  path: '/five-tips-to-master-a-programming-language/',
  modified: '2019-09-01T00:00:00.000Z' }
```

**Comment by u12206050** [comment]
Ah, I found the error within the code. index.js line 131, ID is used
instead of objectID, I'll be able to fix it on Monday. Sorry about that,
have no idea how that got in there, probably after the latest refactoring.

**Comment by u12206050** [comment]
@Edmund1645 It is published now. Fixed in version 2.1.3
